// antd-mobile5/es/components/picker-view/picker-view.js
import React from 'react';
import { usePropsValue } from '../../utils/use-props-value';
import { mergeProps } from '../../utils/with-default-props';
import { Column } from './column';
import { useColumns } from './use-columns';
import { withNativeProps } from '../../utils/native-props';
import { usePickerValueExtend } from './use-picker-value-extend';
var classPrefix = "adm-picker-view";
var defaultProps = {
  defaultValue: []
};
export var PickerView = function PickerView(p) {
  var props = mergeProps(defaultProps, p);

  var _usePropsValue = usePropsValue(Object.assign(Object.assign({}, props), {
    onChange: function onChange(val) {
      var _a;

      (_a = props.onChange) === null || _a === void 0 ? void 0 : _a.call(props, val, generateValueExtend(val));
    }
  })),
      value = _usePropsValue[0],
      setValue = _usePropsValue[1];

  var columns = useColumns(props.columns, value);
  var generateValueExtend = usePickerValueExtend(columns);
  return withNativeProps(props, /*#__PURE__*/React.createElement("div", {
    className: "" + classPrefix
  }, columns.map(function (column, index) {
    return /*#__PURE__*/React.createElement(Column, {
      key: index,
      column: column,
      value: value[index],
      onSelect: function onSelect(val) {
        var nextValue = [].concat(value);
        nextValue[index] = val;
        setValue(nextValue);
      }
    });
  }), /*#__PURE__*/React.createElement("div", {
    className: classPrefix + "-mask " + classPrefix + "-mask-top"
  }), /*#__PURE__*/React.createElement("div", {
    className: classPrefix + "-mask " + classPrefix + "-mask-bottom"
  })));
};
