// @alitajs/plugin-layout/lib/layout/renderRightContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = renderRightContent;

var _react = _interopRequireDefault(require("react"));

var _antd = require("antd");

var _umi = require("umi");

var _icons = require("@ant-design/icons");

require("./style.less");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function renderRightContent(runtimeLayout, loading, initialState, setInitialState) {
  if (runtimeLayout.rightRender) {
    return runtimeLayout.rightRender(initialState, setInitialState, runtimeLayout);
  }

  var menu = /*#__PURE__*/_react.default.createElement(_antd.Menu, {
    className: "umi-plugin-layout-menu"
  }, /*#__PURE__*/_react.default.createElement(_antd.Menu.Item, {
    key: "logout",
    onClick: function onClick() {
      return runtimeLayout.logout && (runtimeLayout === null || runtimeLayout === void 0 ? void 0 : runtimeLayout.logout(initialState));
    }
  }, /*#__PURE__*/_react.default.createElement(_icons.LogoutOutlined, null), "\u9000\u51FA\u767B\u5F55"));

  var avatar = /*#__PURE__*/_react.default.createElement("span", {
    className: "umi-plugin-layout-action"
  }, /*#__PURE__*/_react.default.createElement(_antd.Avatar, {
    size: "small",
    className: "umi-plugin-layout-avatar",
    src: initialState && initialState.avatar || 'https://gw.alipayobjects.com/zos/antfincdn/XAosXuNZyF/BiazfanxmamNRoxxVxka.png',
    alt: "avatar"
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "umi-plugin-layout-name"
  }, initialState && initialState.name));

  if (loading) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "umi-plugin-layout-right"
    }, /*#__PURE__*/_react.default.createElement(_antd.Spin, {
      size: "small",
      style: {
        marginLeft: 8,
        marginRight: 8
      }
    }));
  }

  return initialState && /*#__PURE__*/_react.default.createElement("div", {
    className: "umi-plugin-layout-right anticon"
  }, runtimeLayout.logout ? /*#__PURE__*/_react.default.createElement(_antd.Dropdown, {
    overlay: menu,
    overlayClassName: "umi-plugin-layout-container"
  }, avatar) : avatar, _umi.SelectLang && /*#__PURE__*/_react.default.createElement(_umi.SelectLang, null));
}
