// antd-mobile5/cjs/components/virtual-input/virtual-input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VirtualInput = void 0;

var _react = _interopRequireWildcard(require("react"));

var _nativeProps = require("../../utils/native-props");

var _withDefaultProps = require("../../utils/with-default-props");

var _usePropsValue2 = require("../../utils/use-props-value");

var _classnames = _interopRequireDefault(require("classnames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var classPrefix = 'adm-virtual-input';
var defaultProps = {
  defaultValue: ''
};
var VirtualInput = /*#__PURE__*/(0, _react.forwardRef)(function (p, ref) {
  var _classNames;

  var props = (0, _withDefaultProps.mergeProps)(defaultProps, p);

  var _usePropsValue = (0, _usePropsValue2.usePropsValue)(props),
      value = _usePropsValue[0],
      setValue = _usePropsValue[1];

  var rootRef = (0, _react.useRef)(null);

  var _useState = (0, _react.useState)(false),
      keyboardVisible = _useState[0],
      setKeyboardVisible = _useState[1];

  (0, _react.useLayoutEffect)(function () {
    var root = rootRef.current;
    if (!root) return;

    if (document.activeElement !== root) {
      return;
    }

    root.scrollTo({
      left: root.clientWidth
    });
  }, [value]);
  (0, _react.useImperativeHandle)(ref, function () {
    return {
      focus: function focus() {
        var _a;

        (_a = rootRef.current) === null || _a === void 0 ? void 0 : _a.focus();
      },
      blur: function blur() {
        var _a;

        (_a = rootRef.current) === null || _a === void 0 ? void 0 : _a.blur();
      }
    };
  });

  function onBlur() {
    var _a;

    setKeyboardVisible(false);
    (_a = props.onBlur) === null || _a === void 0 ? void 0 : _a.call(props);
  }

  return (0, _nativeProps.withNativeProps)(props, /*#__PURE__*/_react["default"].createElement("div", {
    ref: rootRef,
    className: (0, _classnames["default"])(classPrefix, (_classNames = {}, _classNames[classPrefix + "-disabled"] = props.disabled, _classNames)),
    tabIndex: props.disabled ? undefined : 0,
    onFocus: function onFocus() {
      var _a;

      setKeyboardVisible(true);
      (_a = props.onFocus) === null || _a === void 0 ? void 0 : _a.call(props);
    },
    onBlur: onBlur,
    onClick: props.onClick
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: classPrefix + "-content"
  }, value, /*#__PURE__*/_react["default"].createElement("div", {
    className: classPrefix + "-caret-container"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: classPrefix + "-caret"
  }))), !value && /*#__PURE__*/_react["default"].createElement("div", {
    className: classPrefix + "-placeholder"
  }, props.placeholder), props.keyboard && /*#__PURE__*/_react["default"].cloneElement(props.keyboard, {
    onInput: function onInput(v) {
      setValue(value + v);
    },
    onDelete: function onDelete() {
      setValue(value.slice(0, -1));
    },
    visible: keyboardVisible,
    onClose: onBlur
  })));
});
exports.VirtualInput = VirtualInput;
