// antd-mobile/cjs/components/virtual-input/virtual-input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VirtualInput = void 0;

var _react = _interopRequireWildcard(require("react"));

var _withDefaultProps = require("../../utils/with-default-props");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var classPrefix = 'adm-virtual-input';
var defaultProps = {
  value: ''
};
var VirtualInput = /*#__PURE__*/(0, _react.forwardRef)(function (p, ref) {
  var props = (0, _withDefaultProps.mergeProps)(defaultProps, p);
  var rootRef = (0, _react.useRef)(null);
  (0, _react.useLayoutEffect)(function () {
    var root = rootRef.current;
    if (!root) return;

    if (document.activeElement !== root) {
      return;
    }

    root.scrollTo({
      left: root.clientWidth
    });
  }, [props.value]);
  (0, _react.useImperativeHandle)(ref, function () {
    return {
      focus: function focus() {
        var _a;

        (_a = rootRef.current) === null || _a === void 0 ? void 0 : _a.focus();
      },
      blur: function blur() {
        var _a;

        (_a = rootRef.current) === null || _a === void 0 ? void 0 : _a.blur();
      }
    };
  });
  return /*#__PURE__*/_react["default"].createElement("div", {
    ref: rootRef,
    className: classPrefix,
    tabIndex: 0,
    onFocus: props.onFocus,
    onBlur: props.onBlur,
    onClick: props.onClick
  }, /*#__PURE__*/_react["default"].createElement("span", {
    className: classPrefix + "-content"
  }, props.value), /*#__PURE__*/_react["default"].createElement("div", {
    className: classPrefix + "-caret-container"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: classPrefix + "-caret"
  })), !props.value && /*#__PURE__*/_react["default"].createElement("span", {
    className: classPrefix + "-placeholder"
  }, props.placeholder));
});
exports.VirtualInput = VirtualInput;
