// antd-mobile5/cjs/components/picker-view/picker-view.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PickerView = void 0;

var _react = _interopRequireDefault(require("react"));

var _usePropsValue2 = require("../../utils/use-props-value");

var _withDefaultProps = require("../../utils/with-default-props");

var _column = require("./column");

var _useColumns = require("./use-columns");

var _nativeProps = require("../../utils/native-props");

var _usePickerValueExtend = require("./use-picker-value-extend");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var classPrefix = "adm-picker-view";
var defaultProps = {
  defaultValue: []
};

var PickerView = function PickerView(p) {
  var props = (0, _withDefaultProps.mergeProps)(defaultProps, p);

  var _usePropsValue = (0, _usePropsValue2.usePropsValue)(Object.assign(Object.assign({}, props), {
    onChange: function onChange(val) {
      var _a;

      (_a = props.onChange) === null || _a === void 0 ? void 0 : _a.call(props, val, generateValueExtend(val));
    }
  })),
      value = _usePropsValue[0],
      setValue = _usePropsValue[1];

  var columns = (0, _useColumns.useColumns)(props.columns, value);
  var generateValueExtend = (0, _usePickerValueExtend.usePickerValueExtend)(columns);
  return (0, _nativeProps.withNativeProps)(props, /*#__PURE__*/_react["default"].createElement("div", {
    className: "" + classPrefix
  }, columns.map(function (column, index) {
    return /*#__PURE__*/_react["default"].createElement(_column.Column, {
      key: index,
      column: column,
      value: value[index],
      onSelect: function onSelect(val) {
        var nextValue = [].concat(value);
        nextValue[index] = val;
        setValue(nextValue);
      }
    });
  }), /*#__PURE__*/_react["default"].createElement("div", {
    className: classPrefix + "-mask " + classPrefix + "-mask-top"
  }), /*#__PURE__*/_react["default"].createElement("div", {
    className: classPrefix + "-mask " + classPrefix + "-mask-bottom"
  })));
};

exports.PickerView = PickerView;
