// antd-mobile5/es/components/virtual-input/virtual-input.js
import React, { forwardRef, useImperativeHandle, useLayoutEffect, useRef, useState } from 'react';
import { withNativeProps } from '../../utils/native-props';
import { mergeProps } from '../../utils/with-default-props';
import { usePropsValue } from '../../utils/use-props-value';
import classNames from 'classnames';
var classPrefix = 'adm-virtual-input';
var defaultProps = {
  defaultValue: ''
};
export var VirtualInput = /*#__PURE__*/forwardRef(function (p, ref) {
  var _classNames;

  var props = mergeProps(defaultProps, p);

  var _usePropsValue = usePropsValue(props),
      value = _usePropsValue[0],
      setValue = _usePropsValue[1];

  var rootRef = useRef(null);

  var _useState = useState(false),
      keyboardVisible = _useState[0],
      setKeyboardVisible = _useState[1];

  useLayoutEffect(function () {
    var root = rootRef.current;
    if (!root) return;

    if (document.activeElement !== root) {
      return;
    }

    root.scrollTo({
      left: root.clientWidth
    });
  }, [value]);
  useImperativeHandle(ref, function () {
    return {
      focus: function focus() {
        var _a;

        (_a = rootRef.current) === null || _a === void 0 ? void 0 : _a.focus();
      },
      blur: function blur() {
        var _a;

        (_a = rootRef.current) === null || _a === void 0 ? void 0 : _a.blur();
      }
    };
  });

  function onBlur() {
    var _a;

    setKeyboardVisible(false);
    (_a = props.onBlur) === null || _a === void 0 ? void 0 : _a.call(props);
  }

  return withNativeProps(props, /*#__PURE__*/React.createElement("div", {
    ref: rootRef,
    className: classNames(classPrefix, (_classNames = {}, _classNames[classPrefix + "-disabled"] = props.disabled, _classNames)),
    tabIndex: props.disabled ? undefined : 0,
    onFocus: function onFocus() {
      var _a;

      setKeyboardVisible(true);
      (_a = props.onFocus) === null || _a === void 0 ? void 0 : _a.call(props);
    },
    onBlur: onBlur,
    onClick: props.onClick
  }, /*#__PURE__*/React.createElement("div", {
    className: classPrefix + "-content"
  }, value, /*#__PURE__*/React.createElement("div", {
    className: classPrefix + "-caret-container"
  }, /*#__PURE__*/React.createElement("div", {
    className: classPrefix + "-caret"
  }))), !value && /*#__PURE__*/React.createElement("div", {
    className: classPrefix + "-placeholder"
  }, props.placeholder), props.keyboard && /*#__PURE__*/React.cloneElement(props.keyboard, {
    onInput: function onInput(v) {
      setValue(value + v);
    },
    onDelete: function onDelete() {
      setValue(value.slice(0, -1));
    },
    visible: keyboardVisible,
    onClose: onBlur
  })));
});
